### PR TITLE
[ticket/17326] Add font icon field for custom BBCodes

### PR DESCRIPTION
**Pull request by danieltj27**
Adds a new field to the custom BBCode add/edit form that allows admins to define a Font Awesome icon to be displayed instead of the BBCodes name on the posting.php editor page.

PHPBB-17326

Checklist:

- [X] Correct branch: master for new features; 3.3.x for fixes
- [x] Tests pass
- [X] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/master/coding-guidelines.html) and [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html)
- [X] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

Tracker ticket:

https://tracker.phpbb.com/browse/PHPBB-17326

### CODE REVIEW

**Comment by danieltj27** [comment]
Third times the charm... please? 😭 

This PR supersedes PRs #6630 and #6627 as it addresses multiple issues flagged in previous commits. Hopefully if the tests pass, this can actually be merged into master/4.0...

Several things over the past three pulls that have been added/address after reviews:
✅ Added core functionality (the BBCode font icon, the whole point of these PRs)
✅ Fix the name of the branch and pull request to adhere to all other pulls
✅ Added an external link (now updated for FA6) to the list of available icons
✅ Use consistent language to describe the icon (ie. BBCode icon instead of button icon)
✅ Use latest Twig syntax for everything (languages, icons and whatever else)
✅ Included new appearance section on the BBCode add/edit form
✅ Set the new column name to be a var char 50 max character limiting column
✅ Added icon preview to reflect similar behaviour in the Pages extension

I think that's the highlights.

Please let me know if I've missed something or made a mistake somewhere and I'll try and fix it. I hope I've done everything right this time. I've tried to cover all bases possible.

File: phpBB/adm/style/acp_bbcodes.html [hunk 1]
```diff
@@ -47,6 +47,17 @@ <h1>{L_ACP_BBCODES}</h1>
 	</dl>
 	</fieldset>
 
+	<fieldset>
+		<legend>{{ lang('APPEARANCE') }}</legend>
+	<dl>
+		<dt><label for="bbcode_font_icon">{{ lang('BBCODE_FONT_ICON') }}</label><br /><span>{{ lang('BBCODE_FONT_ICON_EXPLAIN') }}</span></dt>
+		<dd>
+			<input type="text" name="bbcode_font_icon" id="bbcode_font_icon" value="{{ BBCODE_FONT_ICON }}" />
+			{{ Icon('font', BBCODE_FONT_ICON, '', false, '', {id:"bbcode_icon_preview"}) }}
```
[the remark this answers is not in the document]
Comment (danieltj27) [reply phpBB/adm/style/acp_bbcodes.html:56]: It doesn't actually matter, it works without but I can do that.

File: phpBB/adm/style/admin.js [hunk 2]
```diff
@@ -261,5 +261,10 @@ function parse_document(container)
 		$('.actions a:has(i.acp-icon)').mouseover(function () {
 			$(this).css("text-decoration", "none");
 		});
+
+		// Live update BBCode font icon preview
+		$('#bbcode_font_icon').on('keyup', function(e) {
+			$('#bbcode_icon_preview').attr('class', "o-icon o-icon-font fa-fw fas icon fa-" + $('#bbcode_font_icon').val());
+		});
```
[the remark this answers is not in the document]
Comment (danieltj27) [reply phpBB/adm/style/admin.js:268]: What's the benefit of this? As the JS used here is so small, I don't believe caching the variable would have any material impact on the interface at all. I'm not convinced this is necessary.

File: phpBB/adm/style/acp_bbcodes.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (danieltj27) [reply phpBB/adm/style/acp_bbcodes.html]: Fixed in 5ba1c78

File: phpBB/adm/style/admin.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (iMattPro) [reply phpBB/adm/style/admin.js:268]: You can do that. But to answer the question, you're going and querying the entire DOM again looking for an element that you already found. That is unnecessary.

[the remark this answers is not in the document]
Comment (danieltj27) [reply phpBB/adm/style/admin.js:268]: I've updated this to use `this` in beabaa4. Hopefully that's everything done 🤞

**Pull request closed without merging**